File: PasswordWallet.Client/src/app/mappers/credential.mapper.ts
import { CredentialRequest } from '../contracts/requests/credential-request';
import { CredentialResponse } from '../contracts/responses/credential-response';
import { Credential } from '../models/credential';
import { PaginatedList } from '../models/paginated-list';

export class CredentialMapper {
  public static dtoToModel = (source: CredentialResponse): Credential => {
    return {
      id: source.id,
      username: source.username,
      hiddenPassword: '*********',
      webAddress: source.webAddress,
      description: source.description,
      position: source.position,
    };
  };

  public static toModel = (
    source: PaginatedList<CredentialResponse>
  ): PaginatedList<Credential> => {
    return {
      pageNumber: source.pageNumber,
      pageSize: source.pageSize,
      totalCount: source.totalCount,
      items: source.items.map(this.dtoToModel),
    };
  };

  public static toDto = (source: Credential): CredentialRequest => {
    return {
      username: source.username,
      password: source.password,
      webAddress: source.webAddress,
      description: source.description,
    };
  };
}

File: PasswordWallet.Server/src/Api/Endpoints/v1/Credentials/CredentialMapper.cs
using Api.Endpoints.v1.Credentials.AddCredential;
using Api.Endpoints.v1.Credentials.UpdateCredential;
using Core.Entities;
using Core.Models;
using FastEndpoints;

namespace Api.Endpoints.v1.Credentials;

public class CredentialMapper : ResponseMapper<CredentialResponse, Credential>
{
    public override CredentialResponse FromEntity(Credential credential) => new(credential.Id,
        credential.Username, credential.WebAddress, credential.Description, credential.Position);

    public PaginatedList<CredentialResponse> FromEntities(PaginatedList<Credential> paginatedList) => new(
        paginatedList.PageNumber, paginatedList.PageSize, paginatedList.TotalCount,
        FromEntities(paginatedList.Items).ToList());

    public Credential ToEntity(AddCredentialRequest req) => new()
    {
        Username = req.Username,
        Password = req.Password,
        WebAddress = req.WebAddress,
        Description = req.Description,
        FolderId = req.FolderId
    };

    public Credential ToEntity(UpdateCredentialRequest req) => new()
    {
        Id = req.CredentialId,
        Username = req.Username,
        Password = req.Password,
        WebAddress = req.WebAddress,
        Description = req.Description
    };

    private static IEnumerable<CredentialResponse> FromEntities(IEnumerable<Credential> credentials) =>
        credentials.Select(c => new CredentialResponse(c.Id, c.Username, c.WebAddress, c.Description, c.Position));
}